File: packages/ts-moose-lib/src/utilities/sql.ts
/**
 * Quote a ClickHouse identifier with backticks if not already quoted.
 * Backticks allow special characters (e.g., hyphens) in identifiers.
 */
export const quoteIdentifier = (name: string): string => {
  const trimmed = name.trim();
  return trimmed.startsWith("`") && trimmed.endsWith("`") ?
      trimmed
    : `\`${trimmed}\``;
};

File: packages/py-moose-lib/moose_lib/utilities/sql.py
def quote_identifier(name: str) -> str:
    """Quote a ClickHouse identifier with backticks if not already quoted.

    Backticks allow special characters (e.g., hyphens) in identifiers.
    """
    trimmed = name.strip()
    if trimmed.startswith("`") and trimmed.endswith("`"):
        return trimmed
    return f"`{trimmed}`"


